ip/flac.c

```diff
@@ -427,11 +415,14 @@ static int flac_seek(struct input_plugin_data *ip_data, double offset)
 
 	sample = (uint64_t)(offset * (double)sf_get_rate(ip_data->sf) + 0.5);
 	if (!F(seek_absolute)(priv->dec, sample)) {
+		if (F(get_state(priv->dec)) == FLAC__STREAM_DECODER_SEEK_ERROR) {
+			if (!F(flush)(priv->dec))
+				d_print("failed to flush\n");
+			priv->buf_rpos = 0;
+			priv->buf_wpos = 0;
+		}
 		return -IP_ERROR_ERRNO;
 	}
-	priv->ignore_next_write = 1;
-	priv->buf_rpos = 0;
-	priv->buf_wpos = 0;
```

Comment:
I don't think you've answered my question. I'm talking about our buffer `flac_private::buf`. Should we not clear this buffer before seeking?